### ISSUE TITLE
Analyzer creates error in code for fixed array

### ISSUE THREAD

**Issue by grant-d** [dotnet/roslyn#30483]
**V2017 v 15.8.7**: 

When formatting a document, one of the analyzers (and/or `.editorconfig`) changes the code so as to make it invalid.

I am using a nested class for memory marshalling. It has a `fixed` array declared in it. 

**Steps to Reproduce**:

1. Paste the following code anywhere, even as a nested class
2. Format the document (Ctrl-E, Ctrl-D)
```csharp
private unsafe struct Byte20
    {  // Example bad formatting
    private fixed byte _bytes[20]; // Note fixed modifier
}
```

**Expected Behavior**:
The document is formatted.
```csharp
private unsafe struct Byte20
{  // Fixed bad formatting
    private fixed byte _bytes[20];
}
```

**Actual Behavior**:
The `fixed` modifier is removed and the `readonly` modifier is added, making it invalid.
```csharp
private unsafe struct Byte20
{  // Fixed bad formatting
    private readonly byte _bytes[20]; // Note readonly
}
```
`Compiler Error CS0650: Bad array declarator`
`Compiler Error CS0270: Array size cannot be specified in a variable declaration`

**Workaround**
Though it's not practical - formatting a document is supposed to be innocuous.
```csharp
private unsafe struct Byte20
{
#pragma warning disable IDE0044 // Add readonly modifier
    private fixed byte _bytes[20];
#pragma warning restore IDE0044 // Add readonly modifier
}
```

**Proposed Fix**
Analyzer should detect if a field is a fixed array, and not action it in such a manner.

**Other info**
```xml
<PackageReference Include="Microsoft.Net.Compilers" Version="2.9.0" PrivateAssets="All" />
<PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.2" PrivateAssets="All" />
```

**Comment by sharwell** [comment]
Duplicate of #26364
